src/umpire/strategy/DynamicPool.cpp

```diff
@@ -180,28 +180,27 @@ void* DynamicPool::allocate(std::size_t bytes)
 
     m_curr_bytes += rounded_bytes;
 
-    const int64_t left_bytes{static_cast<int64_t>(
-        free_size - rounded_bytes)};
+    const std::size_t left_bytes{free_size - rounded_bytes};
 
-    if (left_bytes > m_align_bytes) {
+    if (left_bytes > 0) {
       insertFree(static_cast<unsigned char*>(ptr) + rounded_bytes, left_bytes,
                  false, whole_bytes);
     }
   } else {
     // Allocate new block -- note this does not check whether alignment is met
     const std::size_t alloc_bytes{std::max(rounded_bytes, m_min_alloc_bytes)};
+    std::cout << "Allocating " << alloc_bytes << " from resource for " << rounded_bytes << std::endl;
```

Comment:
I assume we want to remove this right?

Reply:
Done already.

Reply:
sorry, I don't no why my browser sometimes keeps old crap around